PV260_Project/BusinessLayer/Services/Implementation/FileUrlService.cs

```diff
@@ -0,0 +1,47 @@
+﻿using DataLayer;
+using DataLayer.Models;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BusinessLayer.Services.Implementation
+{
+    public class FileUrlService : IFileUrlService
+    {
+        private readonly StockSystemDbContext _context;
+
+        public FileUrlService(StockSystemDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<FileUrl?> GetLatest()
+        {
+            return await _context.FileUrls.Where(fileUrl => fileUrl.ValidTo == null).FirstOrDefaultAsync();
+        }
+
+        public async Task<FileUrl> SetNewFileUrl(string url)
+        {
+            var transaction = await _context.Database.BeginTransactionAsync(isolationLevel: System.Data.IsolationLevel.Serializable);
+
+            await _context.Database.ExecuteSqlRawAsync("UPDATE FileUrls SET ValidTo = datetime('now') WHERE ValidTo IS NULL");

```

Comment:
IMO using raw sql is not pretty nice, wouldn't it be better,  if the entities were updated directly?

Reply:
And I got another question: why is the `ValidTo` being set everytime? The property does not have to be nullable and it can be assigned to current time in this method.

Reply:
> IMO using raw sql is not pretty nice, wouldn't it be better, if the entities were updated directly?

Yes, we should use them only if there are performance issues. ORM is much more readable.